src/ddscxx/src/org/eclipse/cyclonedds/pub/qos/DataWriterQosDelegate.cpp

```diff
@@ -33,98 +35,111 @@ namespace qos
 
 DataWriterQosDelegate::DataWriterQosDelegate()
 {
-    reliability_.kind(dds::core::policy::ReliabilityKind::RELIABLE);
+    ddsc_qos(&ddsi_default_qos_writer);
+    present() &= ~QP_DATA_REPRESENTATION;
+    check();
+    policy(dds::core::policy::Reliability(dds::core::policy::ReliabilityKind::RELIABLE));
```

Comment:
Isn't this covered by `ddsc_qos(&ddsi_default_qos_writer)`?

Reply:
Let me double check.

Reply:
As we just use ddsc_qos, and not ddsi_xqos_mergein_missing, we do need to "unset" the DataRepresentation policy.
Setting of the Reliability policy is not necessary, as that set through ddsc_qos, i will remove this.